### PR TITLE
docs: add warning in CSS in JS guide about CSS injection order

### PR DESCRIPTION
**Pull request by bestickley**
#### Description of changes
Add warning in CSS in JS guide about CSS injection order importance as user's custom styles could be overridden.

#### Issue #, if available
Fixes https://github.com/aws-amplify/amplify-ui/issues/3533

#### Description of how you validated changes
Viewed updated markdown locally and ran tests.
image

#### Checklist

- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ cp ./docs/.env.example ./docs/.env
 
 Then set values to the variables.
 
+Note, if you're not using `nvm`, you'll need to use Node.js 16 and set env var with: `export NODE_OPTIONS=--openssl-legacy-provider`
```
Comment (reesscot) [inline docs/README.md:26]: @bestickley Which version of Node 16 are you using? I haven't found I had to set that environment variable unless I was on Node 18

File: docs/src/pages/[platform]/guides/css-in-js/web.mdx [hunk 2]
```diff
@@ -1,9 +1,13 @@
-import { Tabs, TabItem } from '@aws-amplify/ui-react';
+import { Alert, Link, Tabs, TabItem } from '@aws-amplify/ui-react';
 
 One of the goals of Amplify is to be a good citizen of the environment it exists in. As such Amplify works hard to not interfere with other tools that are being used.
 
 There are many CSS In JS libraries out there that give a number of benefits. Amplify does not use a CSS In JS library to implement its theming system but operates well with the major CSS In JS libraries out there. Below are a few of the libraries with descriptions of how they work with Amplify's theming system.
 
+<Alert variation="warning" role="none">
+  If your CSS library depends on injecting style tags into the DOM, make sure they are being injected <b>after</b> Amplify UI's style tag. In other words, ensure the style tag for Amplify UI is listed above your custom styles in the head tag. If Amplify UI's styles are injected after, then your custom styles will likely be unset. See <Link href="https://github.com/aws-amplify/amplify-ui/issues/3533">here</Link> for more info.
```
Comment (reesscot) [inline docs/src/pages/[platform]/guides/css-in-js/web.mdx:8]: We don't have precendent for linking to specific issue from the docs. What do you think of this?

```suggestion
  If your CSS library depends on injecting style tags into the DOM, make sure they are being injected <b>after</b> Amplify UI's style tag. In other words, ensure the style tag for Amplify UI is listed above your custom styles in the `<head>` tag. If Amplify UI's styles are injected after custom styles, then your custom styles may be unset.
```

Comment (bestickley) [reply docs/src/pages/[platform]/guides/css-in-js/web.mdx]: Understood. This looks good to me!

File: docs/src/pages/[platform]/guides/css-in-js/web.mdx [hunk 3]
```diff
@@ -1,9 +1,13 @@
-import { Tabs, TabItem } from '@aws-amplify/ui-react';
+import { Alert, Link, Tabs, TabItem } from '@aws-amplify/ui-react';
```
Comment (ioanabrooks) [inline docs/src/pages/[platform]/guides/css-in-js/web.mdx:1]: nit: Link import is not used

Comment (bestickley) [reply docs/src/pages/[platform]/guides/css-in-js/web.mdx]: Fixed.

**Review by ioanabrooks — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**